catalog/app/utils/tracking.js

```diff
@@ -21,7 +23,22 @@ const consoleTracker = Promise.resolve({
   track: (evt, opts) => console.log(`track: ${evt}`, opts),
 })
 
-const mkLocation = (l) => `${l.pathname}${l.search}${l.hash}`
+function useMkLocation() {
+  const {
+    paths: { passChange: passChangePath },
+    urls: { passChange: passChangeUrl },
+  } = NamedRoutes.use()
+  return React.useCallback(
+    (l) => {
+      const pathname = matchPath(l.pathname, { path: passChangePath, exact: true })
+        ? passChangeUrl('REDACTED')
+        : l.pathname
+
+      return `${pathname}${l.search}${l.hash}`
```

Comment:
I think it doesn't really matter in this place but you can create an output string with URL constructor because `urls.passChange` can occasionally contain `search`. And, it seems that search param can leak here before you reset it in PassChange (but it's from my naive perspective, I didn't dig into it)

Reply:
in this specific case it's safe, bc token is passed in pathname, not as a querystring parameter